zenoh/tests/handler.rs

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+#[test]
+fn pubsub_with_ringbuffer() {
+    use zenoh::{handlers::RingBuffer, prelude::sync::*};
+
+    let zenoh = zenoh::open(Config::default()).res().unwrap();
+    let sub = zenoh
+        .declare_subscriber("test/ringbuffer")
+        .with(RingBuffer::new(3))
+        .res()
+        .unwrap();
+    for i in 0..10 {
+        zenoh
+            .put("test/ringbuffer", format!("put{i}"))
+            .res()
+            .unwrap();
+    }
```

Comment:
For timing resistance I'd add a sleep of 1 second here.

Reply:
Agree